packages/suite-desktop/src/Main.tsx

```diff
@@ -73,13 +73,22 @@ export const init = async (root: HTMLElement) => {
 
     await desktopApi.handshake();
 
-    // render more complex interactive loader with theme provider without redux
-    render(<ModulesLoadingScreen />, root);
+    // Loading Tor as separate module, before the rest of the modules.
+    const { shouldRunTor } = await desktopApi.loadTorModule();
+
+    // When we run this first time `shouldRunTor` will tell if Tor should run according to previous settings,
+    // when it runs because of renderer (e.g. Ctrl+R) it will always be false.
+    if (shouldRunTor) {
+        await new Promise(resolve => {
+            // Switching Tor on but not waiting, instead we wait for events from
+            // Tor ControlPort `tor/bootstrap` inside TorLoadingScreen.
+            desktopApi.toggleTor(true);
```

Comment:
Isn't it possible (although very very improbable) that all the events from `toggleTor` trigger before they're listened to from `<TorLoadingScreen />`? And if it is, shouldn't it be resolved by calling toggleTor right **after** `render(<TorLoadingScreen />)`?

Reply:
It is indeed very improbable but why not switch those 2 lines and have it formally correct?

Reply:
Indeed, very improbably but better to have the lines in the right place. :+1:  eb0c46f71628afa11992cfec493cd7400f4bb341